sdk/tables/azure-data-tables/inc/azure/data/tables/tables_clients.hpp

```diff
@@ -374,31 +375,70 @@ namespace Azure { namespace Data { namespace Tables {
      * @param options Optional client options that define the transport pipeline policies for
      * authentication, retries, etc., that are applied to every request.
      */
-    explicit TableServicesClient(
+    explicit TableServiceClient(
         const std::string& serviceUrl,
         std::shared_ptr<Azure::Data::Tables::Credentials::SharedKeyCredential> credential,
         const TableClientOptions& options = {});
+
+    /**
+     * @brief Initializes a new instance of tableClient.
+     *
+     * @param serviceUrl A url referencing the table that includes the name of the account and the
+     * name of the table.
+     * @param credential The shared key credential used to sign requests.
+     * @param options Optional client options that define the transport pipeline policies for
+     * @param sasBuilder The shared access signature builder used to sign requests.
+     * authentication, retries, etc., that are applied to every request.
+     */
+    explicit TableServiceClient(
+        const std::string& serviceUrl,
+        std::shared_ptr<Azure::Data::Tables::Credentials::SharedKeyCredential> credential,
+        Azure::Data::Tables::Sas::AccountSasBuilder& sasBuilder,
```

Comment:
The sasBuilder is a mutable reference? That's surprising.

Can it be a const reference? And what happens if the sasBuilder's lifetime is different from that of the TableServiceClient? C++ won't catch that kind of lifetime mismatch.